fe/fe-core/src/test/java/com/starrocks/sql/plan/AggregateTest.java

```diff
@@ -1882,4 +1884,113 @@ public void testGroupByLiteral() throws Exception {
         assertContains(plan, "  3:Project\n" +
                 "  |  <slot 3> : -9223372036854775808");
     }
+
+    @Test
+    public void testRewriteSumByAssociativeRule() throws Exception {
+        // 1. different types
+        // 1.1 nullable
+        String sql = "select sum(t1b+1),sum(t1c+1),sum(t1d+1),sum(t1e+1),sum(t1f+1),sum(t1g+1),sum(id_decimal+1)" +
+                " from test_all_type";
```

Comment:
what about `-`,`+`,`/`?

Reply:
> what about `-`,`*`,`/`?

* `-` is supported now. 
* I will support `*` in other PR later. 
* `/` may introduce precision problems, I need to confirm whether we can rewrite it like this